index.mjs

```diff
@@ -1,10 +1,11 @@
+const midCache = new Map(); // midpoint vertices cache to avoid duplicating shared vertices
 
 export default function icomesh(order = 4) {
     if (order > 10) throw new Error(`Max order is 10, but given ${order}.`);
 
     // set up an icosahedron (12 vertices / 20 triangles)
     const f = (1 + Math.sqrt(5)) / 2;
-    const T = Math.pow(4, order);
+    const T = 1 << (order << 1); // Math.pow(4, order);
```

Comment:
No need to micro-optimize something that happens once — readability is more important.